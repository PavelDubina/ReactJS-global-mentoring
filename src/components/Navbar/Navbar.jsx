import React, { useState } from 'react'
import className from 'classnames'
import { Select } from './Select/Select'
import { navGenres } from '../../utils/constants'
import styles from './Navbar.scss'

export const Navbar = () => {
  const [active, setActive] = useState(0)
  return (
    <nav className={styles.container}>
      <div className={styles.genres}>
        {navGenres.map((genre, i) => {
          const classList = className(styles.genre, { [styles.active]: active === i })
          return (
            <div className={classList} onClick={() => setActive(i)} key={genre}>
              {genre}
            </div>
          )
        })}
      </div>
      <div className={styles.sort}>
        <span>SORT BY</span>
        <Select />
      </div>
    </nav>
  )
}
