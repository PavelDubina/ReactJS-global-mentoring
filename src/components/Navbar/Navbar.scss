.container {
  position: relative;
  height: 44px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  font-size: 20px;

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 3px;
    background-color: #555;
  }
}

.genre {
  position: relative;
  margin: 0 10px;
  cursor: pointer;
}

.genres {
  display: flex;
  text-transform: uppercase;
  color: #fff;

  & .genre:nth-child(1) {
    margin-left: 0;
  }
}

.active {
  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 3px;
    background-color: #f65261;
    z-index: 1;
  }
}

.sort {
  display: flex;

  & span {
    color: #ffffff80;
  }

  & div {
    height: 24px;
    display: flex;
    align-items: center;
    margin: 0 0 0 24px;
    color: #fff;
    cursor: pointer;

    &::after {
      content: '\25BC';
      margin: 0 0 0 10px;
      font-size: 12px;
      color: #f65261;
    }
  }
}
