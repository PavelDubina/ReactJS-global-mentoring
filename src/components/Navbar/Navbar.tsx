import React from 'react'
import className from 'classnames'
import { useTranslation } from 'next-i18next'
import { Select } from './Select/Select'
import { navGenres } from '../../utils/constants'
import styles from '../../../styles/Navbar.module.scss'

type NavbarProps = {
  genreValue: string
  sortValue: string
  toggleSortValue: (sortValue: string) => void
  handleMenu: (genre: string) => void
}

export const Navbar: React.FC<NavbarProps> = ({ genreValue, sortValue, toggleSortValue, handleMenu }) => {
  const { t } = useTranslation()
  return (
    <nav className={styles.container}>
      <div className={styles.genres}>
        {Object.values(navGenres).map((genre) => {
          const classList = className(styles.genre, { [styles.active]: genreValue === genre })
          return (
            <div className={classList} onClick={() => handleMenu(genre)} key={genre}>
              {t(`nav.genres.${genre}`)}
            </div>
          )
        })}
      </div>
      <div className={styles.sort}>
        <span>{t('nav.sort.by')}</span>
        <Select value={sortValue} toggleSortValue={toggleSortValue} />
      </div>
    </nav>
  )
}
