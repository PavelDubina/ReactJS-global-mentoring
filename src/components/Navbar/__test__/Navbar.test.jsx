import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Navbar } from '../Navbar'

describe('Navbar', () => {
  const getComponent = ({
    genreValue = 'all',
    sortValue = 'release_date',
    toggleSortValue = jest.fn(),
    handleMenu = jest.fn(),
  }) =>
    render(
      <Navbar
        handleMenu={handleMenu}
        toggleSortValue={toggleSortValue}
        sortValue={sortValue}
        genreValue={genreValue}
      />,
    )
  it('Should renders correctly', () => {
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('handleMenu have been called and elenement has class active', () => {
    const handleMenu = jest.fn()
    const { getByText } = getComponent({ handleMenu })
    expect(getByText('all')).toBeTruthy()
    fireEvent.click(getByText('all'))
    expect(handleMenu).toBeCalled()
    expect(getByText('all').classList.contains('active')).toBeTruthy()
  })
})
