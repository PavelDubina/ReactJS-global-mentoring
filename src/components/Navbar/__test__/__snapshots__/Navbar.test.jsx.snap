// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="container"
  >
    <div
      class="genres"
    >
      <div
        class="genre active"
      >
        all
      </div>
      <div
        class="genre"
      >
        documentary
      </div>
      <div
        class="genre"
      >
        comedy
      </div>
      <div
        class="genre"
      >
        horror
      </div>
      <div
        class="genre"
      >
        crime
      </div>
    </div>
    <div
      class="sort"
    >
      <span>
        SORT BY
      </span>
      <div
        class="select"
        data-testid="select"
      >
        release_date
        <div
          class="dropdown"
          data-testid="dropdown"
        >
          <p
            class=""
            data-testid="RELEASE DATE"
          >
            RELEASE DATE
          </p>
          <p
            class=""
            data-testid="RATING"
          >
            RATING
          </p>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
